### PR TITLE
Split Article Java Basics

### LINKED ISSUE #797: Splitting Articles
@alayek, @Rafase282 and @theflametrooper need to split all these articles into multiple ones which are smaller:

- [x] https://github.com/FreeCodeCamp/wiki/blob/master/Python-Basics.md
- [x] https://github.com/FreeCodeCamp/wiki/blob/master/Python-Functions.md **Waiting for Merge**
- [x] https://github.com/FreeCodeCamp/wiki/blob/master/Python-More-Builtin-Types.md **Waiting for Merge**
- [x] https://github.com/FreeCodeCamp/wiki/blob/master/Web-Development-in-Python.md **Waiting for Merge**
- [x] https://github.com/FreeCodeCamp/wiki/blob/master/Java-Loops.md **Waiting for Merge**
- [x] https://github.com/FreeCodeCamp/wiki/blob/master/Java-Basics.md **Waiting for Merge**

### PR DESCRIPTION
**Pull request by Rafase282**
Splits the Java article into smaller articles, some have to-dos

Closes #797 

If possible merge this one last so it will close the issue.

### CODE REVIEW

File: Java-Basic-Inheritance.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
+# Java Inheritance
+
+So great you have successfully created a Car class. But, wait, aren't Tesla cars supposed to be electric variants? I want an Electric car class, but it also should have the properties of the original `Car` class.
```
Comment (alayek) [inline Java-Basic-Inheritance.md]: > So great you have successfully created a Car class.

Maybe phrase it differently, with proper reference to where they had created the `Car` class?

**Comment by koustuvsinha** [comment]
@Rafase282 possible camperbot issue : when people call "help java inheritance" then the main article about inheritance should come up. but no one will call "help java basics inheritance" , instead the logical call should be "help java inheritancd basics". Thus you should rename the files from Java-Basics-Inheritance to Java-Inheritance-Basics, and same for interfaces. @alayek @abhisekp verify if this would solve the issue

**Comment by abhisekp** [comment]
@Rafase282 
@koustuvsinha is right. :white_check_mark: 
Lets rename the file name from ~~`Java-Basic-Inheritance.md`~~ to `Java-Inheritance.md` or `Java-Inheritance-Basic.md` as suggested by @koustuvsinha

**Comment by abhisekp** [comment]
Lets just drop `Basic` thingy.

Inheritance is _just_ Inheritance.
Interface is _just_ Interface. 

Why the basic suffix/prefix? What's your opinion @koustuvsinha ?

**Comment by koustuvsinha** [comment]
@abhisekp Java-Inheritance.md already exists having a deeper details about Inheritance. Same for Interfaces. here since we are splitting the basics page, thats why those are Inheritance basics and Interfaces basics.

If its creating too confusion, i would suggest drop these two files and link to original Inheritance / Interface. what say guys?

**Comment by abhisekp** [comment]
@koustuvsinha umm... no dropping. :-1: 
Let it be Inheritance Basics then :+1: :+1:

**Comment by abhisekp** [comment]
@Rafase282 @koustuvsinha Lets use this at the end of all the pages for navigating.

----

[**Previous**](Java-Intro) | [**Home**](Java) | [**Next**](Java-Inheritance)
:---|:---:|---:
<kbd><a href="Java-Intro"> :point_left: <strong>Java Intro</strong></a></kbd> | <kbd><a href="java"> :earth_asia: <strong>Java Home (TOC)</strong> :earth_asia: </a></kbd> | <kbd><a href="Java-Inheritance"><strong>Java Inheritance :point_right: </strong></a></kbd>

----

```md
[**Previous**](Java-Intro) | [**Home**](Java) | [**Next**](Java-Inheritance)
:---|:---:|---:
<kbd><a href="Java-Intro"> :point_left: <strong>Java Intro</strong></a></kbd> | <kbd><a href="java"> :earth_asia: <strong>Java Home (TOC)</strong> :earth_asia: </a></kbd> | <kbd><a href="Java-Inheritance"><strong>Java Inheritance :point_right: </strong></a></kbd>
```

**Comment by abhisekp** [comment]
@Rafase282 **File Rename**
```diff
- Java-Loops-Do-While.md
+ Java-do-while-Loop.md
```

and also other such modifications i.e.

- use **`lowercase`** for Java keywords
- use obvious naming convention i.e. _**for loop**_ instead of _**loop for**_, etc

**Comment by koustuvsinha** [comment]
haha cool @abhisekp! thats great

**Comment by abhisekp** [comment]
@Rafase282 Also add a blank heading 2 after first para or **_short_** intro text for your test of luv :heart: towards @camperbot 

```md
##
```

File: Java-Basic-Interfaces.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        return name;
+    }
+      // Getter method
+    public String getManufacturerName() {
+        return manufacturerName;
+    }
+}
+```
+
+:rocket: [REPL It!](https://repl.it/CJZa/0)
```
Comment (abhisekp) [inline Java-Basic-Interfaces.md]: @Rafase282 @koustuvsinha how about this for repl it demo links?

----
<kbd><a href="https://repl.it/CJZa/0"> <strong>REPL it!</strong> :rocket: </a></kbd>

----
```md
<kbd><a href="https://repl.it/CJZa/0"><img src="https://i.imgur.com/80UCMbk.png"> <strong>REPL it!</strong> :rocket: </a></kbd>
```

**Comment by alayek** [comment]
@abhisekp awesome!

**Comment by Rafase282** [comment]
The naming convention I'm using is helpful when people are searching and they don't know the exact name. If they do **for loop** it will show, yet they can also transition from Java to Loops, to any time of loops easily with the naming convention.

**Comment by Rafase282** [comment]
@abhisekp I mostly just separated the content, didn't do much editing other than adding link for empty sections or adding a brief description to the original article so it wouldn't just be a bunch of links.

File: Java-Basic-Interfaces.md (hunk 2 above)
Comment (koustuvsinha) [inline Java-Basic-Interfaces.md]: yeah looks awesome @abhisekp ! :+1:

**Comment by Rafase282** [comment]
With regards to the new navigation, Lets leave that for a separated PR. If we are going to use that then we should add it to all similar articles not just these.

Also keep in mind the html or people will start adding their own html css again to wiki articles.

**Comment by koustuvsinha** [comment]
@Rafase282 sorry but nitpicking :stuck_out_tongue_closed_eyes: 

```diff
- Java-Inheritance-Basic.md
+ Java-Inheritance-Basics.md
```

**Comment by Rafase282** [comment]
[**Previous**](Java-Intro) | [**Home**](Java) | [**Next**](Java-Inheritance-Basics)
:---|:---:|---:
<a href="Java-Intro"><kbd> :point_left: <strong>Java Intro</strong></kbd> </a>| <kbd><a href="java"> :earth_asia: <strong>Java Home (TOC)</strong> :earth_asia: </a></kbd> | <kbd><a href="Java-Inheritance-Basics"><strong>Java Inheritance Basics :point_right: </strong></a></kbd>

@abhisekp do you see the difference now?

**Comment by abhisekp** [comment]
@Rafase282 how about this?

<a href="Java-Intro"><kbd> :point_left: <strong> Java Intro</strong></kbd></a> | <kbd><a href="java"> :earth_asia: <strong>Java Home (TOC)</strong> :earth_asia: </a></kbd> | <kbd><a href="Java-Inheritance-Basics"><strong>Java Inheritance Basics :point_right: </strong></a></kbd>
:---|:---:|---:

**Comment by Rafase282** [comment]
With the rectangle as the link:

[**Previous**](Java-Intro) | [**Home**](Java) | [**Next**](Java-Inheritance-Basics)
:---|:---:|---:
<a href="Java-Intro"><kbd> :point_left: <strong>Java Intro</strong></kbd> </a>| <a href="java"><kbd> :earth_asia: <strong>Java Home (TOC)</strong> :earth_asia: </kbd></a> | <a href="Java-Inheritance-Basics"><kbd><strong>Java Inheritance Basics :point_right: </strong></kbd></a>

With just the text as the link:

[**Previous**](Java-Intro) | [**Home**](Java) | [**Next**](Java-Inheritance-Basics)
:---|:---:|---:
<kbd><a href="Java-Intro"> :point_left: <strong>Java Intro</strong></a></kbd> | <kbd><a href="java"> :earth_asia: <strong>Java Home (TOC)</strong> :earth_asia: </a></kbd> | <kbd><a href="Java-Inheritance-Basics"><strong>Java Inheritance Basics :point_right: </strong></a></kbd>

If the problem is the link color that can be changed but I personally don't have a problem with the no colored links.

**Comment by abhisekp** [comment]
# Proposal 1 :ring: 

<div align="center">
<kbd><a href="Java-Intro">:point_left: <strong> Java Intro</strong><br>Previous</a></kbd> &mdash; <kbd align="center"><a href="java"> :earth_asia: <strong>Java Home</strong> :earth_asia: <br>Table of Contents</a></kbd> &mdash; <kbd align="right"><a href="Java-Inheritance-Basics"><strong>Java Inheritance Basics</strong> :point_right:<br>Next</a></kbd>
</div>

**Comment by abhisekp** [comment]
# Proposal 2 :ring: 

<div align="center">
<a href="Java-Intro"><kbd>:point_left: <strong> Java Intro</strong><br>Previous</kbd></a> &mdash; <a href="java"><kbd align="center"> :earth_asia: <strong>Java Home</strong> :earth_asia: <br>Table of Contents</kbd></a> &mdash; <a href="Java-Inheritance-Basics"><kbd align="right"><strong>Java Inheritance Basics</strong> :point_right:<br>Next</kbd></a>
</div>

**Comment by abhisekp** [comment]
# Proposal 3 :ring: 

[**Previous**](Java-Intro) | [**Home**](Java) | [**Next**](Java-Inheritance-Basics)
:---|:---:|---:
<kbd><a href="Java-Intro"> :point_left: <strong>Java Intro</strong></a></kbd> | <kbd><a href="java"> :earth_asia: <strong>Java Home (TOC)</strong> :earth_asia: </a></kbd> | <kbd><a href="Java-Inheritance-Basics"><strong>Java Inheritance Basics :point_right: </strong></a></kbd>

**Comment by abhisekp** [comment]
# Proposal 4

[**Previous**](Java-Intro) | [**Home**](Java) | [**Next**](Java-Inheritance-Basics)
:---|:---:|---:
<a href="Java-Intro"><kbd> :point_left: <strong>Java Intro</strong></kbd> </a>| <a href="java"><kbd> :earth_asia: <strong>Java Home (TOC)</strong> :earth_asia: </kbd></a> | <a href="Java-Inheritance-Basics"><kbd><strong>Java Inheritance Basics :point_right: </strong></kbd></a>

**Comment by abhisekp** [comment]
# Proposal 5 :ring: 

<div align="center">
<a href="Java-Intro"><kbd>:point_left: <strong> Java Intro</strong></kbd></a> &mdash; <a href="java"><kbd align="center">:earth_asia: <strong>Java Home</strong> :earth_asia:</kbd></a> &mdash; <a href="Java-Inheritance-Basics"><kbd align="right"><strong>Java Inheritance Basics</strong> :point_right:</kbd></a>
</div>

**Comment by abhisekp** [comment]
# Proposal 6 :ring: 

<div align="center">
<kbd><a href="Java-Intro">:point_left: <strong> Java Intro</strong></a></kbd> &mdash; <kbd align="center"><a href="java">:earth_asia: <strong>Java Home</strong> :earth_asia:</a></kbd> &mdash; <kbd align="right"><a href="Java-Inheritance-Basics"><strong>Java Inheritance Basics</strong> :point_right:</a></kbd>
</div>

**Comment by Rafase282** [comment]
I like proposal 4 or proposal 2 in that order.

**Comment by koustuvsinha** [comment]
# LGTM :+1:

**Comment by theflametrooper** [comment]
## LGTM :+1:, just waiting for @alayek

**Pull request merged**